### PR TITLE
feat: check if WAF logging is enabled

### LINKED ISSUE #8: Config rule - Ensure WAF ACL logs are replicating to satellite s3 bucket

### PR DESCRIPTION
**Pull request by mohamed-cds**
- Custom AWS config rule to check if WAF v2 ACL is set to log to either S3 or Kinesis
- Setup central account assume role permissions for accounts added to the `satellite_accounts.json`
- Remove SSEAlgorithm in the s3 access log config rule since it's unsupported

Closes #8

### CODE REVIEW

File: terragrunt/aws/config/wafv2_logs.tf [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+#
+data "archive_file" "cbs_wafv2_logs_rule" {
+  type        = "zip"
+  source_file = "config_rules/wafv2_logs/webacl_logs_rule.py"
+  output_path = "/tmp/cbs_wafv2_logs_rule.zip"
+}
+
+resource "aws_lambda_function" "cbs_wafv2_logs_rule" {
+  filename      = "/tmp/cbs_wafv2_logs_rule.zip"
+  function_name = "cbs_wafv2_logs_rule"
```
Comment (patheard) [inline terragrunt/aws/config/wafv2_logs.tf:38]: Can you change this to either have a `Cbs` or `cbs-` prefix (for the AWS nuke config exemptions).

File: terragrunt/aws/config/wafv2_logs.tf [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+    (var.billing_tag_key) = var.billing_tag_value
+    Terraform             = true
+  }
+}
+
+#
+# Lambda execution role
+#
+resource "aws_iam_role" "cbs_wafv2_logs_rule" {
+  name               = "cbs_wafv2_logs_rule"
```
Comment (patheard) [inline terragrunt/aws/config/wafv2_logs.tf:82]: Same comment about name prefix with a `Cbs` or `cbs-`.

Also totally fine adding a `cbs_` prefix exemption though to our AWS nuke config.

File: terragrunt/aws/config/wafv2_logs.tf [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role_policy_attachment" "cbs_wafv2_logs_rule_wafv2_list_web_acls" {
+  role       = aws_iam_role.cbs_wafv2_logs_rule.name
+  policy_arn = aws_iam_policy.wafv2_list_web_acls.arn
+}
+
+resource "aws_iam_policy" "wafv2_list_web_acls" {
+  name        = "wafv2_list_web_acls"
```
Comment (patheard) [inline terragrunt/aws/config/wafv2_logs.tf:102]: Just needs a `Cbs` of `cbs-` prefix.

File: terragrunt/aws/config/wafv2_logs.tf [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+    }
+  }
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
+module "waf_logs" {
```
Comment (patheard) [inline terragrunt/aws/config/wafv2_logs.tf:160]: I'm assuming we'll want a replication rule on this resource that sends the logs to the satellite s3 bucket?

File: terragrunt/aws/config/wafv2_logs.tf [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+      enabled = true
+      expiration = {
+        days = 14
+      }
+    }
+  ]
+}
+
+resource "aws_iam_role" "waf_log_role" {
+  name               = "${var.account_id}-logs"
```
Comment (patheard) [inline terragrunt/aws/config/wafv2_logs.tf:180]: AWS nuke tweak: `Cbs` or `cbs-` prefix

File: terragrunt/aws/config/wafv2_logs.tf [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+  assume_role_policy = data.aws_iam_policy_document.firehose_assume_role.json
+
+  tags = {
+    CostCentre = var.billing_code
+    Terraform  = true
+  }
+}
+
+resource "aws_iam_policy" "write_waf_logs" {
+  name        = "${var.account_id}_WriteLogs"
```
Comment (patheard) [inline terragrunt/aws/config/wafv2_logs.tf:190]: AWS nuke tweak: `Cbs` or `cbs-` prefix

**Review by patheard — CHANGES_REQUESTED** [review]
I like this compliance/remediate combo in a single Lamda.  Much cleaner than managing an entirely separate thing.

Had a few changes around AWS nuke config exempltion naming, but also totally open to adjusting those configs if we want to support the `cbs_` naming prefix as well.

File: terragrunt/aws/config/wafv2_logs.tf (hunk 4 above)
Comment (mohamed-cds) [reply terragrunt/aws/config/wafv2_logs.tf:160]: @patheard i was curious about this. Will your rule automatically do this ? I didn't want to duplicate aws config rules so i removed the replication to the satellite bucket.

Comment (patheard) [reply terragrunt/aws/config/wafv2_logs.tf:160]: My thinking had been that there's a lot of config and IAM involved in replicating cross-account with a CMK, so I wanted to limit that setup to only one S3 bucket.

**Review by patheard — DISMISSED** [review]
I'd say we can merge this and iterate as we figure out how we want to manage replication of WAF to the log archive account.

File: terragrunt/aws/config/wafv2_logs.tf (hunk 4 above)
Comment (mohamed-cds) [reply terragrunt/aws/config/wafv2_logs.tf]: refactored to replicate to the satellite bucket.

File: terragrunt/aws/config/inputs.tf [hunk 7]
```diff
@@ -4,3 +4,9 @@ variable "config_max_execution_frequency" {
   default     = "TwentyFour_Hours"
   type        = string
 }
+
+variable "billing_code" {
```
Comment (patheard) [inline terragrunt/aws/config/inputs.tf:8]: Could you use the common `var.billing_tag_value` for this instead?

File: terragrunt/aws/satellite_bucket/iam.tf [hunk 8]
```diff
@@ -46,7 +46,8 @@ data "aws_iam_policy_document" "s3_replicate" {
       "s3:ReplicateDelete"
     ]
     resources = [
-      "${local.log_archive_bucket_arn}/*"
+      "${local.log_archive_bucket_arn}/*",
+      "${module.satellite_bucket.s3_bucket_arn}/*"
```
Comment (patheard) [inline terragrunt/aws/satellite_bucket/iam.tf:50]: I think we can get rid of this if our test works.

**Review by patheard — APPROVED** [review]
🎉

**Pull request merged**